rosetta/docs/GPU_performance.md

```diff
@@ -0,0 +1,131 @@
+    - --xla_gpu_all_gather_combine_threshold_bytes=8589934592
+    - --xla_gpu_reduce_scatter_combine_threshold_bytes=8589934592
+      
+- Combine threshold values in XLA that determine when an all-reduce (AR) is triggered. Typically, used to overlap AR of gradients with back-prop of compute. We want to set this to be atleast as large as possible to achieve high efficiency, but as small as possible to achieve maximum overlap. Depending on the interconnect of your system, one might want to try several threshold values in steps of 2 from say 16MB to total gradient size.
+    - --xla_gpu_all_reduce_combine_threshold_bytes=8589934592
+
+
+### Flags for Collective permute 
+
+The following flags enable overlap of pipeline parallel communication of send/recv with computation. 
+- --xla_gpu_enable_pipelined_p2p=true  (false by default)
+- --xla_gpu_collective_permute_decomposer_threshold=33521664
+
+### Flags to enable Collective Matmul
+
+The following flags enable overlap of tensor parallel communication with GEMMs/matmul by splicing GEMMs into smaller chunks and triggering each chunks' collective right after the chunk's GEMM is done. The threshold determines the size of output buffer of GEMM when this optimization becomes active (0 enables collective matmul for all GEMM-collective patterns)
+- --xla_gpu_multi_streamed_windowed_einsum=true
+- --xla_gpu_threshold_for_windowed_einsum_mib=0
+
+### Profile Guided Latency Estimator (PGLE)
+
+The following flag enables use of PGLE with JAX/XLA. Please see [PGLE notes](https://github.com/NVIDIA/JAX-Toolbox/blob/main/rosetta/docs/PGLE.md) for more details.
+- --xla_gpu_pgle_profile_file_or_directory_path=filename
+
+## Other XLA Flags 
+
+### Enable CUDA graphs
+
+The below enables CUDA Graph suppport for JAX/XLA workloads
+- --xla_gpu_enable_command_buffer (1 by default, available options are 0, 1, 2, and 3)
+
+### Dynamic-Update Slice Fusion
+
+The following flag removes extra copies introduced by DUS (dynamic update slice) when used in conjunction with custom NVIDIA kernels (like cuBLAS for GEMMs)
+- --xla_gpu_enable_custom_fusions=true
+
+### NCCL Optimizations
+
+Enable user-buffers in NCCL for zero-copy collectives and send/recv. Needs NCCL_NVLS_ENABLE=1 for AG, AR, RS.
+- --xla_gpu_enable_nccl_user_buffers=true
+
+Flags to improves memory consumed by NCCL.
+- --xla_gpu_enable_nccl_resource_sharing=true  
+- --xla_gpu_enable_nccl_per_stream_comms=false [https://github.com/openxla/xla/pull/9845](https://github.com/openxla/xla/pull/9845)
+
+Fine-grain control to improve performance by initializing a NCCL communicators to use only max_nchannels (SMs) 
+- --xla_gpu_nccl_collective_max_nchannels
+- --xla_gpu_nccl_p2p_max_nchannels
+
+### Debug flags 
+- --xla_dump_to=some/path
+- --xla_dump_latency_hiding_schedule=true
+- --xla_gpu_lhs_enable_gpu_async_tracker=false
```

Comment:
What is the async tracker?

Reply:
I think I left this here mistakenly.
This is used to PP comm overlap without circular repeat. Ill move it to the right location.